### ISSUE TITLE
Making an ajax call to load data before layout collapse trigger? 

### ISSUE THREAD

**Issue by davidawad** [olifolkerd/tabulator#1421]
Hey oli, as always, thank you so much for tabulator, it's fantastic.

Quick question for you; 
Looking at the tabulator docs there's a collapsable menu option [here](http://tabulator.info/docs/3.5?#responsive-layout-collapse). Is it possible to make a request, or somehow run some code when this button is clicked to add additional information? Essentially If I click a row(`1` out of `n`) that row could correspond to a lot more information. (say `m` total key / value pairs) 

Basically I don't want to make one request for `m`*`n` data, I want to make a request for n rows, and fill in the m key/val pairs when someone clicks the button to see more information / collapse the menu. 

I know that *_responsiveLayoutCollapseFormatter_* is out there based on the documentation and I suppose this would be what I would use, but will it get held up by making ajax requests in the formatter when it gets clicked? or is there a better way? thanks.

**Comment by olifolkerd** [comment]
Hey @davidawad 

Easy peasy, you wold use a ***rowFormatter*** to add a holding element for this extra data:
```js
rowFormatter:function(row){
     //row - row component
    row.append("<div class='row-subcomponent'></div>");       
},
```

and then either add an extra column with a button that is triggered on a click event, or for sake of simplicity for this example im going to use a ***rowClick*** event to the trigger the loading of the data:

```js
rowClick:function(e, row){
    // make an ajax request or process some data

    //create some HTML for the new ro display and lets assign it to the newEls variable

    //append the new elemen to the row sub component element
    $(".row-subcomponent", row.getElement()).append(newEls);
}
```

you could even include a nested table if you want, an example of this can be [Found Here](http://tabulator.info/examples/3.5#nested-tables)

I hope that helps,

Cheers

Oli :)
